src/ucp/am/rndv.c

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2022, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ucp_am.inl"
+
+#include <ucp/proto/proto_single.inl>
+#include <ucp/rndv/proto_rndv.inl>
+
+
+static size_t ucp_am_rndv_rts_pack(void *dest, void *arg)
+{
+    ucp_request_t *req          = arg;
+    ucp_rndv_rts_hdr_t *rts_hdr = dest;
+    size_t rts_size, max_bcopy, total_size;
+
+    ucp_am_fill_header(ucp_am_hdr_from_rts(rts_hdr), req);
+    rts_hdr->opcode = UCP_RNDV_RTS_AM;
+    rts_size        = ucp_proto_rndv_rts_pack(req, rts_hdr, sizeof(*rts_hdr));
+    if (req->send.msg_proto.am.header_length == 0) {
+        return rts_size;
+    }
+
+    total_size = rts_size + req->send.msg_proto.am.header_length;
+    max_bcopy  = ucp_ep_get_max_bcopy(req->send.ep, req->send.lane);
+    if (ucs_unlikely(total_size > max_bcopy)) {
+        ucs_fatal("RTS is too big %lu, max %lu", total_size, max_bcopy);
+    }
```

Comment:
not sure we need it: the transport that calls this function would already check if size exceeds limits. and in any case, better check it in ucp_am_send_nbx, compare to the max user header size that ucp_worker_query() would return, and only if ENABLE_PARAMS_CHECK is 1
so can also remove the if in line 25